simpleast-core/src/test/java/com/discord/simpleast/code/JavaScriptRuleTest.kt

```diff
@@ -0,0 +1,221 @@
+        """
+          /** Multiline
+              Comment
+          */
+        """.trimIndent(),
+        "// Inlined",
+        "// Line comment")
+  }
+
+  @Test
+  fun strings() {
+    val ast = parser.parse("""
+      ```js
+      x = 'Hello';
+      y = "world";
+      log(`Hi`);
+      ```
+    """.trimIndent(), TestState())
+
+    ast.assertNodeContents<StyleNode.TextStyledNode<*>>(
+        "'Hello'",
+        "\"world\"",
+        "`Hi`")
+  }
+  @Test
+  fun stringsMultiline() {
+    val ast = parser.parse("""
+      ```js
+      text = `
+      hello
+      world
+      `;
+      ```
+    """.trimIndent(), TestState())
+
+    ast.assertNodeContents<StyleNode.TextStyledNode<*>>(
+        """
+          `
+          hello
+          world
+          `
+        """.trimIndent(),
+    )
+  }
+
+  @Test
+  fun functions() {
+    val ast = parser.parse("""
+      ```js
+      function test(T) {
+        // Implementation
+      }
+      fn = function () {};
+      ```
+    """.trimIndent(), TestState())
+
+    ast.assertNodeContents<JavaScript.FunctionNode<*>>(
+      "function test(T)",
+      "function ()")
+  }
+
+  @Test
+  fun commentedFunction() {
+    val ast = parser.parse("""
+      ```js
+      /*
+        function test(T) {
+          throw new Error();
+        }
+      */
+      // function O() {}
+      log(x /* test var */);
+      ```
+    """.trimIndent(), TestState())
+
+    ast.assertNodeContents<StyleNode.TextStyledNode<*>>(
+        """
+        /*
+          function test(T) {
+            throw new Error();
+          }
+        */
+        """.trimIndent(),
+        "// function O() {}",
+        "/* test var */")
+  }
+
+  @Test
+  fun keywords() {
+    val ast = parser.parse("""
+      ```js
+      while (true) {}
+      for (;;) {}
+      if (false) {}
+      class {}
+      try {
+      } catch (err) {
+        throw
+      } finally {
+        return;
+      }
+      ```
+    """.trimIndent(), TestState())
+    ast.assertNodeContents<StyleNode.TextStyledNode<*>>(
+        "while", "true", 
+        "for", "if", "false",
+        "class", "try", "catch",
+        "finally", "return", "throw")
+  }
+
+  @Test
+  fun numbers() {
+    val ast = parser.parse("""
+      ```js
+      x = 0;
+      x += 69;
+      max(x, 420);
+      ```
+    """.trimIndent(), TestState())
+    ast.assertNodeContents<StyleNode.TextStyledNode<*>>(
+            "0", "69", "420"
+   )
+  }
+
+  @Test
+  fun fields() {
+    val ast = parser.parse("""
+      ```js
+      var foo = x;
+      let bar = y;
+      const baz = z;
+        ```
+    """.trimIndent(), TestState())
+    ast.assertNodeContents<JavaScript.FieldNode<*>>(
+        "var foo",
+        "let bar",
+        "const baz")
+  }
+
+  @Test
+  fun classDef() {
+    val ast = parser.parse("""
+      ```js
+      class Bug {}
+      ```
+    """.trimIndent(), TestState())
+
+    ast.assertNodeContents<CodeNode.DefinitionNode<*>>(
+        "class Bug")
+  }
+
```

Comment:
I believe we are missing the `object property` and `generic` test cases.

Reply:
The generic test is already there, object property test was missing which i added :)